### PR TITLE
Idempotent migration director1 -> director2

### PR DESCRIPTION
**Pull request by jochenschneider**

### CODE REVIEW

**Comment by jochenschneider** [comment]
I've added and index which seems to make the ecu_targets migration go way faster.

**Comment by jochenschneider** [comment]
Updated.

File: src/main/resources/db/migration/V2__migrate_data_from_director1.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+join director.current_images c on e.namespace = c.namespace and e.ecu_serial = c.ecu_serial
+join director2.ecu_targets t on t.namespace = e.namespace AND t.filename = c.filepath AND t.checksum = c.checksum
+on duplicate key update device_id = e.device, public_key = e.public_key, hardware_identifier = e.hardware_identifier, current_target = t.id, created_at = e.created_at, updated_at = e.updated_at;
+
+-- devices <- ecus
+
+insert into director2.devices (namespace,id,primary_ecu_id,created_at,updated_at)
+select namespace,device,ecu_serial,created_at,updated_at
+from director.ecus e
+where e.primary = 1 and exists (select * from director2.ecus e2 where e.namespace = e2.namespace and e.ecu_serial = e2.ecu_serial)
```
Comment (simao) [inline src/main/resources/db/migration/V2__migrate_data_from_director1.sql]: we just skip the ones that don't have ecu?

Comment (jochenschneider) [reply src/main/resources/db/migration/V2__migrate_data_from_director1.sql]: Not sure what you mean. This and the previous table are based on ECUs.

**Pull request merged**